packages/axios-extension/test/abap/ui5-version-service.test.ts

```diff
@@ -0,0 +1,191 @@
+        service.log = mockLogger;
+
+        // Mock the get method
+        mockGet = jest.spyOn(service, 'get');
+    });
+
+    afterEach(() => {
+        jest.clearAllMocks();
+    });
+
+    describe('PATH constant', () => {
+        it('should have the correct base path', () => {
+            expect(UI5VersionService.PATH).toBe('/sap/public/bc/ui5_ui5');
+        });
+    });
+
+    describe('getUI5Version', () => {
+        it('should return UI5 version when API call succeeds', async () => {
+            // Arrange
+            const mockResponse = {
+                data: { Version: '1.96.27' },
+                status: 200,
+                statusText: 'OK',
+                headers: {},
+                config: {}
+            };
+            mockGet.mockResolvedValue(mockResponse);
+
+            // Act
+            const result = await service.getUI5Version();
+
+            // Assert
+            expect(result).toBe('1.96.27');
+            expect(mockGet).toHaveBeenCalledWith('/bootstrap_info.json', {
+                transformResponse: expect.any(Function)
+            });
+        });
+
+        it('should throw UI5VersionError when version is missing', async () => {
+            // Arrange
+            const mockResponse = {
+                data: { Version: '' },
+                status: 200,
+                statusText: 'OK',
+                headers: {},
+                config: {}
+            };
+            mockGet.mockResolvedValue(mockResponse);
+
+            // Act & Assert
+            await expect(service.getUI5Version()).rejects.toThrow('UI5 version not provided.');
+            expect(mockLogger.error).toHaveBeenCalledWith('Could not get UI5 Version.');
+            expect(mockLogger.debug).toHaveBeenCalled();
+        });
+
+        it('should throw UI5VersionError when version is null', async () => {
+            // Arrange
+            const mockResponse = {
+                data: { Version: null },
+                status: 200,
+                statusText: 'OK',
+                headers: {},
+                config: {}
+            };
+            mockGet.mockResolvedValue(mockResponse);
+
+            // Act & Assert
+            await expect(service.getUI5Version()).rejects.toThrow('UI5 version not provided.');
+            expect(mockLogger.error).toHaveBeenCalledWith('Could not get UI5 Version.');
+        });
+
+        it('should throw UI5VersionError when version is undefined', async () => {
+            // Arrange
+            const mockResponse = {
+                data: {},
+                status: 200,
+                statusText: 'OK',
+                headers: {},
+                config: {}
+            };
+            mockGet.mockResolvedValue(mockResponse);
+
+            // Act & Assert
+            await expect(service.getUI5Version()).rejects.toThrow('UI5 version not provided.');
+            expect(mockLogger.error).toHaveBeenCalledWith('Could not get UI5 Version.');
+        });
```

Comment:
It would probably be enough to leave 1-2 tests here because these three are checking the same thing